src/library/tags/tag.h

```diff
@@ -0,0 +1,310 @@
+#pragma once
+
+#include <QJsonValue>
+#include <QVector>
+
+#include "util/assert.h"
+#include "util/macros.h"
+#include "util/math.h"
+#include "util/optional.h"
+
+namespace mixxx {
+
+namespace library {
+
+namespace tags {
+
+/// A category for tags.
+///
+/// Constraints: Lowercase ASCII, no whitespace characters
+class TagFacet final {
```

Comment:
Please split these classes into separate files. I got confused at because I expected this class to be `Tag` matching the name of the file.